[Synthetics] Update tests for restricted roles (#1423)
DataDog/terraform-provider-datadog: docs/resources/synthetics_private_location.md

### Optional

- **description** (String) Description of the private location.

> - **metadata** (Block List, Max: 1) The private location metadata (see [below for nested schema](#nestedblock--metadata))

Author: I don't think we'll be able to change this wording because it's auto-generated.  See [synthetics_test.md](https://github.com/DataDog/terraform-provider-datadog/blob/f89268e55913488cb66812b80d4a514a63472d9d/docs/resources/synthetics_test.md?plain=1) for many examples of this exact phrasing.

- **tags** (List of String) A list of tags to associate with your synthetics private location.

### Read-Only

- **config** (String, Sensitive) Configuration skeleton for the private location. See installation instructions of the private location on how to use this configuration.
- **id** (String) The ID of this resource.

<a id="nestedblock--metadata"></a>
### Nested Schema for `metadata`

Optional:

> - **restricted_roles** (List of String) A list of role identifiers that can be pulled from the Roles API, for restricting read and write access.

Reviewer: ```suggestion
- **restricted_roles** (List of String) A list of role identifiers pulled from the Roles API to restrict read and write access.
```
